firmware/Core/Src/telecommands/telecommand_definitions.c

```diff
@@ -280,3 +304,156 @@ uint8_t TCMDEXEC_available_telecommands(const char *args_str, TCMD_TelecommandCh
+    const uint8_t found_int = CONFIG_find_string_variable_by_name((const char *)config_var_name, strlen((char *)config_var_name), &res);
+    if (found_int > 0)
+    {
+        snprintf(response_output_buf, response_output_buf_len, "%s : %s\n", "Could not find", config_var_name);
+        return 0;
+    }
+
+    char original_value[strlen(res->variable_pointer)];
+    strncpy(original_value, res->variable_pointer, res->max_length);
+    const uint8_t set_result = CONFIG_assign_str_new_value(&res->variable_pointer, config_var_new_value, res->max_length);
+    if (set_result > 0)
+    {
+        snprintf(response_output_buf, response_output_buf_len, "Could not set config var: %s because new value is too long\n", config_var_name);
+        return 0;
+    }
+    
+    snprintf(response_output_buf, response_output_buf_len, "Set config var: %s from \"%s\" to \"%s\" \n", config_var_name, original_value, res->variable_pointer);
+    return 0;
+}
+
+uint8_t TCMDEXEC_get_integer_configuration_variable(const char *args_str, TCMD_TelecommandChannel_enum_t tcmd_channel,
+                                                    char *response_output_buf, uint16_t response_output_buf_len)
+{
+    CONFIG_integer_config_entry_t *res;
+    const uint8_t found_int = CONFIG_find_integer_variable_by_name((const char *)args_str, strlen((char *)args_str), &res);
+    if (found_int > 0)
+    {
+        snprintf(response_output_buf, response_output_buf_len, "%s : %s\n", "Could not find", args_str);
+        return 0;
+    }
+
+    char json_buf[200];
+    memset(json_buf, 0, 200);
+    CONFIG_num_var_to_json(res, json_buf, 200);
+    snprintf(response_output_buf, response_output_buf_len, "%s", json_buf);
+    return 0;
+
+}
+
+uint8_t TCMDEXEC_get_string_configuration_variable(const char *args_str, TCMD_TelecommandChannel_enum_t tcmd_channel, char *response_output_buf, uint16_t response_output_buf_len)
+{
+    CONFIG_string_config_entry_t *res;
+    const uint8_t found_int = CONFIG_find_string_variable_by_name((const char *)args_str, strlen((char *)args_str), &res);
+    if (found_int > 0)
+    {
+        snprintf(response_output_buf, response_output_buf_len, "%s : %s\n", "Could not find", args_str);
+        return 0;
+    }
+
+    char json_buf[200];
+    memset(json_buf, 0, 200);
+    CONFIG_str_var_to_json(res, json_buf, 200);
+    snprintf(response_output_buf, response_output_buf_len, "%s", json_buf);
+    return 0;
+}
+uint8_t TCMDEXEC_get_all_configuration_variables(const char *args_str, TCMD_TelecommandChannel_enum_t tcmd_channel,
+                                                 char *response_output_buf, uint16_t response_output_buf_len)
+{
+
+    const char *header = "Available Configuration Variables:\n\n";
+    const char *integer_header = "Integer Configuration Variables:\n\n";
+    snprintf(response_output_buf, response_output_buf_len, "%s%s", header, integer_header);
+    
+    char num_print_table[200];
+    memset(num_print_table, 0, 200);
```

Comment:
Consider just setting `num_print_table[0] = '\0';` instead of the memset, especially in loops